lang/src/org/partiql/lang/eval/Scalar.kt

```diff
@@ -50,6 +52,20 @@ interface Scalar {
      */
     fun dateValue(): LocalDate? = null
 
+    /**
+     * Returns this value as a [LocalTime] or `null` if not applicable.
+     * This operation is only applicable for [ExprValueType.TIME].
+     * Note that this will return [LocalTime] value even if the timezone offset is specified.
+     * To get the time with timezone value, use [timeWithTimezoneValue] which returns an instance of [OffsetTime]
+     */
+    fun timeValue(): LocalTime? = null
+
+    /**
+     * Returns this value as a [OffsetTime] or `null` if not applicable.
+     * This operation is only applicable for [ExprValueType.TIME] with timezone offset.
+     */
+    fun timeWithTimezoneValue(): OffsetTime? = null
+
```

Comment:
Would it be more appropriate to return a variant type for both of these modalities?  Or if we have to capture precision, we return our time wrapper that has the capability to return these modes.

Reply:
Included a wrapper class that wraps the entire TIME value including `localTime`, `zoneOffset` and `precision`.